cirq/contrib/noise_models/noise_models.py

```diff
@@ -211,54 +212,140 @@ def noisy_moment(self, moment: 'cirq.Moment',
-    specified separately for each qubit.
+class PerQubitDepolarizingWithDampedReadoutNoiseModel(devices.NoiseModel):
+    """DepolarizingWithDampedReadoutNoiseModel which allows depolarization
+    probabilities to be specified separately for each qubit.
 
-    Similar to depol_prob in DepolarizingNoiseModel, depol_prob_map should map
-    Qids in the device to their depolarization probability.
+    This simulates asymmetric readout error. The noise is structured
+    so the T1 decay is applied, then the readout bitflip, then measurement.
+    Note that T1 decay is only applied to measurement, not other gates.
+
+    In moments without measurement, all qubits affected by an operation will
+    have a depolarizing channel applied after the original operation. Qubits
+    that remain idle will be unaffected by this model.
+
+    As with the DepolarizingWithDampedReadoutNoiseModel, if a circuit contains
+    measurements, they must be in moments that don't also contain gates.
     """
 
     def __init__(
             self,
-            depol_prob_map: Dict['cirq.Qid', float],
+            depol_prob_map: Dict['cirq.Qid', float] = None,
+            bitflip_prob_map: Dict['cirq.Qid', float] = None,
+            decay_prob_map: Dict['cirq.Qid', float] = None,
     ):
-        """A depolarizing noise model with variable per-qubit noise.
+        """A depolarizing noise model with damped readout error.
+
+        All error modes are specified on a per-qubit basis. To omit a given
+        error mode from the noise model, leave its map blank when initializing
+        this object.
 
         Args:
             depol_prob_map: Map of depolarizing probabilities for each qubit.
+            bitflip_prob: Probability of a bit-flip during measurement.
+            decay_prob: Probability of T1 decay during measurement.
+                Bitflip noise is applied first, then amplitude decay.
         """
-        for qubit, depol_prob in depol_prob_map.items():
-            value.validate_probability(depol_prob, f'depol prob of {qubit}')
+        for prob_map, desc in [(depol_prob_map, "depolarization prob"),
+                               (bitflip_prob_map, "readout error prob"),
+                               (decay_prob_map, "readout decay prob")]:
+            if prob_map:
+                for qubit, prob in prob_map.items():
+                    value.validate_probability(prob, f'{desc} of {qubit}')
         self.depol_prob_map = depol_prob_map
+        self.bitflip_prob_map = bitflip_prob_map
+        self.decay_prob_map = decay_prob_map
 
     def noisy_moment(self, moment: 'cirq.Moment',
                      system_qubits: Sequence['cirq.Qid']):
-        if (_homogeneous_moment_is_measurements(moment) or
-                self.is_virtual_moment(moment)):
+        if self.is_virtual_moment(moment):
             return moment
+        moments = []
+        if _homogeneous_moment_is_measurements(moment):
```

Comment:
Will this handle the case that the measurements are (purposefully or accidentally) mixed with other types of gates?

It seems like this needs to be handled by either putting amplitude damping in, or for throwing an exception if it is a case we do not support.

Reply:
Currently the `_homogeneous_moment_is_measurements` function raises an exception for moments containing both measurement and non-measurement gates. Once damping is supported for general gates, we can optionally apply it to measurement gates and remove this check.